addons/udes_stock/models/stock_move_line.py

```diff
@@ -861,11 +871,7 @@ def _prepare_task_info(self):
         }
         user_scans = picking.picking_type_id.u_user_scans
 
-        if user_scans == "product":
-            task["type"] = "product"
-            task["pick_quantity"] = sum(self.mapped("product_qty"))
-            task["quant_ids"] = self.get_quants().get_info()
-        else:
+        if user_scans in ["package", "pallet"]:
```

Comment:
I may be over-thinking this, but the way we are using `user_scans` doesn't feel quite right to me: we seem to be using it as a proxy for some unknown abstraction?  I'm not sure what it might be, but it feels (to me) like the task type ought to be computed by the picking type  rather than derived from `user_scans`.  Possibly also the guards on the old behaviour.

Reply:
Yep I get what you're saying. Although not sure what you mean by computed by the picking type? Do you mean we should just be looking at the move line. e.g. if the ML has a product in a package, we should be scanning a package?

Reply:
I mean something like

```python
class StockPickingType(Model):

    @property
    def task_type(self):
        if self.user_scans in ['package', 'pallet']:
            return 'package'
        if self.user_scans:
            return 'package'
        return 'general'
```

assuming that wouldn't break anything...

Reply:
Ah gotcha, let me try it out. :)